backend/arm64/emit.ml

```diff
@@ -2097,6 +2097,13 @@ let fundecl fundecl =
 
 (* Emission of data *)
 
+let int64_to_int32_exn n =
+  if Int64.compare n 0x80000000L (* Int32.int_min *) < 0
+     || Int64.compare n 0x7FFFFFFFL (* Int32.int_max *) > 0
+  then Misc.fatal_error "int64_to_int32_exn: out of range";
+  Int64.to_int32 n
```

Comment:
I would move this to the Numbers module, maybe in an Int64 submodule or something.

Reply:
(also you shouldn't need to hardcode the constants, the min/max values are defined in Int32 I think)

Reply:
+ use Misc.fatal_errorf with the offending `n`